river/command/keyboard.zig

```diff
@@ -58,15 +72,33 @@ pub fn keyboardLayout(
     ) orelse return error.InvalidValue;
     defer new_keymap.unref();
 
-    server.config.keymap.unref();
-    server.config.keymap = new_keymap.ref();
+    applyLayout(new_keymap);
+}
 
-    var it = server.input_manager.devices.iterator(.forward);
-    while (it.next()) |device| {
-        if (device.wlr_device.type != .keyboard) continue;
-        const wlr_keyboard = device.wlr_device.toKeyboard();
-        // wlroots will log an error if this fails and there's unfortunately
-        // nothing we can really do in the case of failure.
-        _ = wlr_keyboard.setKeymap(new_keymap);
-    }
+pub fn keyboardLayoutFile(
+    _: *Seat,
+    args: []const [:0]const u8,
+    _: *?[]const u8,
+) Error!void {
+    var file = std.fs.cwd().openFile(args[1], .{}) catch return error.CannotReadFile;
+    defer file.close();
+
+    var file_bytes = file.readToEndAlloc(util.gpa, std.math.maxInt(usize)) catch |err| {
```

Comment:
Let's cap this at something exceedingly large but not ludicrous like the current value (millions of terrabytes). I suggest one GiB or (`1024 * 1024 * 1024).

It will probably never be relevant, but in the case that some poor user accidentally mistakes their 100 GiB 16k@240 fps video for a keymap file they won't cause their system (likely running with overcommit enabled) to grind to a halt or cause river to get OOM killed.